### DISCUSSION TITLE
Set JWT Expiry maximum to longer than one week

### DISCUSSION THREAD

**Discussion by WJimmyCook** [supabase/supabase#3636] [Feature Requests]
For mobile apps, it doesn't make sense to force the user to log back in after one week without use. Users expect (for certain kinds of apps) that if they open an app after a few weeks they are still logged in. Social media apps, journaling apps, workout apps, etc.

**Comment by Vinzent03** [comment]
The access token (jwt) is only valid for a certain time. But every time you sign in, you get a refresh token too. This is valid forever and can be used to obtain a new access token. I think all client libraries do that automatically. See [this comment](https://github.com/supabase/supabase/discussions/3482#discussioncomment-1456844) for additional explanation.

**Reply by WJimmyCook** [reply]
You're right, now I see that the client is automatically refreshing the access token. Something else must be occasionally causing a log out. I'll have to do some more debugging. Thanks!

**Reply by Vinzent03** [reply]
Maybe this https://github.com/supabase/supabase/discussions/3637 ?

**Reply by WJimmyCook** [reply]
Yep, that seems to be it. Thanks for debugging my app for me! haha

**Reply by Vinzent03** [reply]
No problem. I went exactly the same way as you and found this problem today.
